server.go

```diff
@@ -990,10 +995,30 @@ func (s *server) broadcastMessages(
 func (s *server) SendToPeer(target *btcec.PublicKey,
 	msgs ...lnwire.Message) error {
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	// Queue the incoming messages in the peer's outgoing message buffer.
+	// We acquire the shared lock here to ensure the peer map doesn't change
+	// from underneath us.
+	s.mu.RLock()
+	targetPeer, errChans, err := s.sendToPeer(target, msgs)
+	s.mu.RUnlock()
```

Comment:
Seems better to lock inside s.sendToPeer - given any other caller should lock for that method as well.

Reply:
This is intentional, as an internal method the caller _should_ be aware of the locking context and required granularity. Broadcasting also uses this method, locking inside would cause us to reacquire the lock for each peer in the broadcast.

Reply:
Sorry, broadcasting uses `sendPeerMessages`, not `sendToPeer`. However, `SendToPeer` is the public method that we expect to behave atomically, so it should decide the proper lock granularity. `sendToPeer` could be used internally by other methods that require different granularities

Reply:
Admittedly, it does look a little abnormal because of the pipelining :P